app/src/components/LayoutHeader.vue

```diff
@@ -42,30 +42,138 @@
+            right-2
+            md:right-20
+            bg-white
+            text-grey-400
+            flex-col
+            gap-y-2
+            uppercase
+            font-medium
+            text-left
+            whitespace-nowrap
+          "
+        >
+          <div>{{ userDisplayName }}</div>
+          <div>{{ balance }} ETH</div>
+          <div class="border-b border-grey-400 my-4"></div>
+
+          <button
+            @click="changeWallet"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            change wallet
+          </button>
+          <button
+            @click="disconnect"
+            class="cursor-pointer hover:text-grey-500 flex no-underline uppercase font-medium"
+          >
+            disconnect wallet
+          </button>
+        </div>
       </div>
-    </nav>
+    </div>
+    <div v-else-if="!isSupportedNetwork" class="ml-auto flex items-center">
+      <div class="flex h-12 items-center pr-2 w-12">
+        <WarningIcon class="icon stroke-pink" />
+      </div>
+      <div class="text-gray-500">Unsupported network</div>
+    </div>
+    <div v-else @click="connectWallet" class="flex items-center h-14 gap-x-2 group cursor-pointer ml-auto">
+      <div class="hidden md:block group-hover:text-grey-500">Connect</div>
+      <div>
+        <ConnectWalletIcon class="icon icon-primary" />
+      </div>
+    </div>
   </header>
 </template>
 
 <script lang="ts">
 import { defineComponent } from 'vue';
 import useWalletStore from 'src/store/wallet';
 import { WarningIcon } from '@fusion-icons/vue/interface';
+import { Wallet3Icon as ConnectWalletIcon } from '@fusion-icons/vue/web3';
+import { ArrowBottom2Icon as ArrowBottomIcon } from '@fusion-icons/vue/interface';
+import Jazzicon from 'src/components/Jazzicon.vue';
+import { Cart2Icon as CartIcon } from '@fusion-icons/vue/interface';
+import { formatEther, JsonRpcProvider, Web3Provider } from 'src/utils/ethers';
 
 // Header menu bar items
 const navigation = [
   { name: 'Home', href: '/' },
   { name: 'Cart', href: '/cart' },
-  { name: 'Contact', href: '/contact' },
 ];
 
 export default defineComponent({
   name: 'LayoutHeader',
-  components: { WarningIcon },
+  components: { ConnectWalletIcon, ArrowBottomIcon, CartIcon, Jazzicon, WarningIcon },
   emits: ['toggleAbout'],
+  data() {
+    return {
+      balance: '0',
+      provider: null as Web3Provider | JsonRpcProvider | null,
+    };
+  },
+  watch: {
+    userAddress: 'updateBalance',
+  },
+  methods: {
+    async updateBalance() {
+      if (!this.userAddress) {
+        return;
+      }
+      const balance = await this.provider?.getBalance?.(this.userAddress);
+      this.balance = formatEther(balance ?? '0');
+    },
+    disconnect() {
+      // ToDo implement disconnect
+      console.log('disconnecting');
+    },
+    changeWallet() {
+      // ToDo implement changeWallet
+      console.log('changing wallets');
+    },
```

Comment:
I could use some help implementing these. Is there a method inside useWallet store that I could use?

Reply:
We use the onboard.js package to manage wallet state, so:
- For `disconnect`, use [`onboard.walletReset()`](https://docs.blocknative.com/onboard#walletreset)
- For `changeWallet`, use [`await onboard.walletSelect()`](https://docs.blocknative.com/onboard#walletselect). For this one, it should automatically update the `last-wallet` in local storage when you change wallets (see where we call `setLastWallet`), but if not you'll need to do that also

Reply:
I implemented [disconnectWallet](https://github.com/dcgtc/dgrants/pull/162/files#diff-a870919603ea024ec4349c9ed046dfae2dc99dd8405389e17f9af666c3cd2f00R116-R121). Is it all we need?

Regarding changeWallet, just to be clear: we just want to show the modal with the wallet providers so the user can switch to another one, right? Something like:

```js
await onboard.walletSelect(); 
await onboard.walletCheck();
await configureProvider(); 
// Make sure last-wallet is updated
```

Reply:
> I implemented disconnectWallet. Is it all we need?

Hm, in the linked method, where is `resetState` defined? I can't find what that is. But otherwise that looks right to me, and I think you really only need the `onboard.walletReset()` call

> Regarding changeWallet, just to be clear: we just want to show the modal with the wallet providers so the user can switch to another one, right? Something like:

Yep that is the intention, and yes that snippet looks correct to me, but make sure to test both as confirmation!

Reply:
`resetState` is defined on [wallet.ts](https://github.com/dcgtc/dgrants/blob/main/app/src/store/wallet.ts#L50)

```js
// Reset state when, e.g.user switches wallets. Provider/signer are automatically updated by ethers so are not cleared
function resetState() {
  userAddress.value = undefined;
  network.value = undefined;
}
```

Without calling it the UI wasn't updating properly (Still saw my connected wallet)

Reply:
Oh yes perfect, good call. Sorry I only searched for it in the diff—forgot it was a function we already have